src/controller/python/chip/yaml/runner.py

```diff
@@ -256,6 +351,37 @@ def run_action(self, dev_ctrl: ChipDeviceCtrl) -> _ActionResult:
         return _ActionResult(status=_ActionStatus.ERROR, response=None)
 
 
+class WaitForReportAction(BaseAction):
+    '''Single WaitForReport action to be executed.'''
+
+    def __init__(self, test_step, context: _ExecutionContext):
+        '''Converts 'test_step' to wait for report action.
+
+        Args:
+          'test_step': Step containing information required to run wait for report action.
+          'context': Contains test-wide common objects such as DataModelLookup instance.
+        Raises:
+          UnexpectedParsingError: Raised if the expected queue does not exist.
+        '''
+        super().__init__(test_step.label)
+        self._attribute_name = stringcase.pascalcase(test_step.attribute)
+        self._output_queue = context.subscription_callback_result_queue.get(self._attribute_name,
+                                                                            None)
+        if self._output_queue is None:
+            raise UnexpectedParsingError(f'Could not find output queue')
+
+    def run_action(self, dev_ctrl: ChipDeviceCtrl) -> _ActionResult:
+        try:
+            # This is a blocking call. While there should be a timeout here, the current codegen
```

Comment:
will the global timeout be enforced?
Even then, should we set some timeout here that is quite large but still non-infinite (like 10minutes)?

Would be nice to have some update later on (TODO comment?) to have timeouts on these things.